### PR TITLE
add parameter w

### LINKED ISSUE #20: Parameter -w

### PR DESCRIPTION
**Pull request by justinmueller05082000**
fixes #20

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -w -e 'a *a' <<EOF
a a
a a
EOF
```

... doesn't work as expected.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -F -w -e 'a a' <<EOF
a a
a a
EOF
```

... doesn't work as expected.

File: main.go [hunk 1]
```diff
@@ -40,6 +41,26 @@ func main() {
...
+		}
+	} else if *wordsParameter && *fixedStringsParameter {
+		for _, i := range flagTest {
+			re, regErr := regexp.Compile(regexp.QuoteMeta(i))
+			if regErr != nil {
+				fmt.Fprintln(os.Stderr, regErr)
+				os.Exit(1)
+			}
+			regArr = append(regArr, re)
+		}
```
Comment (Al2Klimov) [inline main.go]: There's already a loop for regexp.Compile()ing. Any way not to duplicate that loop and to just regexp.QuoteMeta() the stuff in the new loop?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -w -e 'a *a' <<EOF
a a
a a
EOF
```

... doesn't work as expected.

File: main.go [hunk 2]
```diff
@@ -40,8 +41,22 @@ func main() {
...
+			if !strings.HasSuffix(flagTest[i], "\\b") {
+				flagTest[i] = flagTest[i] + "\\b"
+			}
+		}
+	}
+
+	if !*fixedStringsParameter || *wordsParameter && *fixedStringsParameter {
 		for _, i := range flagTest {
+			if *wordsParameter && *fixedStringsParameter {
+				i = regexp.QuoteMeta(i)
```
Comment (Al2Klimov) [inline main.go]: Any way to reduce the whole code's -F handling independent of -w to this one in a dedicated loop?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (justinmueller05082000) [reply main.go]: if you say so there probably is one but i dont quite get it

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -w -e 'a|b' <<EOF
a
b
ab
EOF
```

... doesn't work as expected.

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
```bash
go run . -w -e '(a|b)\b' <<EOF
a
b
ab
EOF
```

... doesn't work.

File: main.go [hunk 3]
```diff
@@ -40,17 +40,28 @@ func main() {
...
+	if *fixedStringsParameter {
+		for i := range flagTest {
+			flagTest[i] = regexp.QuoteMeta(flagTest[i])
 		}
 	}
 
+	if *wordsParameter {
+		for i := range flagTest {
+			flagTest[i] = "\\b(" + flagTest[i]
+			flagTest[i] = flagTest[i] + ")\\b"
```
Comment (Al2Klimov) [inline main.go]: Any way to shorten this?

**Review by Al2Klimov — CHANGES_REQUESTED** [review]

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
* [x] `git reset --soft HEAD~$(git rev-list --count $(git merge-base HEAD master)..HEAD^)`
* [x] `git commit --amend --no-edit`
* [x] `git push -f origin feature/add-parameter-w-20`

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Write a test protocol. Cover all cases you changed the algorithm of.

**Comment by justinmueller05082000** [comment]
**Test protocol:**

**toGrep.txt:**
```
foobar cat dog mouse
dog mouse zebra fo*bar
foooooobar elephant cat mouse
zebra raccoon cat dolphin
```

All cases are combined with **`-e`** to work.

________________

**Case (1): `-F`**

**My Code - Terminal:** `go run . -F -e 'fo*bar' < toGrep.txt` 

**Output (1/2):**
```
dog mouse zebra fo*bar
```

**Grep - Terminal:** `grep -F -e 'fo*bar' < toGrep.txt` 

**Output (2/2):**
```
dog mouse zebra fo*bar
```
________________

**Case (2): `-w`**

**My Code - Terminal:** `go run . -w -e 'ze*bra' < toGrep.txt` 

**Output (1/2):**
```
dog mouse zebra fo*bar
zebra raccoon cat dolphin
```
**Grep - Terminal:** `grep -w -e 'ze*bra' < toGrep.txt` 

**Output (2/2):**
```
dog mouse zebra fo*bar
zebra raccoon cat dolphin
```

________________

**Case (3): `-w` & `-F`**

**My Code - Terminal:** `go run . -F -w -e 'fo*bar' < toGrep.txt` 

**Output (1/2):**
```
dog mouse zebra fo*bar
```
**Grep - Terminal:** `grep -F -w -e 'fo*bar' < toGrep.txt` 

**Output (2/2):**
```
dog mouse zebra fo*bar
```

________________

**Result:**
All results are equal except for the colored highlighting in the command line.

**Review by Al2Klimov — APPROVED** [review]

**Pull request merged**